src/py21cmfast/drivers/_param_config.py

```diff
@@ -204,6 +204,9 @@ def _get_inputs(kwargs: dict[str, Any]) -> InputParameters:
         example) and this is fine. So, here we return the inputs of the "most advanced"
         OutputStruct that is given.
         """
+        # TODO: Figure out what happens when the first astro box (HaloBox, IonizedBox or SpinTemp)
+        #   is passed inputs=None, does it pull the astro params from the matter boxes, if so that's bad
+        #   We probably shouldn't ever store/save the astro params in the matter boxes
```

Comment:
Let's turn this into an issue and think about solving it, because I think it's an important point.

Reply:
I forgot that we always require full parameter specification, so this is a minor issue. There are some edge cases since AstroParams are saved to the InitialConditions structs, which means they will just get overwritten with the latest AstroParams (same actual data) and a user might accidentally load in an unexpected parameter set. This should always get caught in the consistency checks though.